thermod/runBARCO.R

```diff
@@ -172,3 +131,53 @@ g4 <- ggplot(output) +
   labs(x = 'Simulated Time', y = 'Fluxes in mg/(cm3 d)')  +
   theme_bw()+
   theme(legend.position="bottom");g4
+
+
+
+### forecasting framework
+
+for (ii in 1:3){
+  print(paste0('run number: ', ii))
+
+  boundary_noise <- add_noise(boundary)
+
+  for (jj in 1:100){
+    y_noise <- yini
+    y_noise[1] <- rnorm(1, mean = y_noise[1], sd = 0.1)
+    y_noise[2] <- rnorm(1, mean = y_noise[2], sd = 100)
+
+    out <- run_oxygen_forecast(bc = boundary_noise, params = wq_parameters,
+                               ini = y_noise, times = times, ice = ice_on)
+
+    result <- data.frame('Time' = time_seq,
+                         'WT_sim' = out[,2],
+                         'DO_sim' = out[,3],
+                         'id' = as.character(ii),
+                         'ini' = as.character(jj))
+    head(result)
+    result_filter = result
+    result_filter$WT_sim = kalman_filtering(time = result_filter$Time, series = result_filter$WT_sim)
+    result_filter$DO_sim = kalman_filtering(time = result_filter$Time, series = result_filter$DO_sim)
```

Comment:
where are the `kalman_filtering()` functions?

Reply:
in the thermod package under R/helper.R. It's mainly used for smoothing the time series.